piecrust/src/session.rs

```diff
@@ -116,6 +117,35 @@ impl Session {
         let module_id = ModuleId::from_bytes(hash.into());
 
         self.deploy_with_id(module_id, bytecode)?;
+
+        Ok(module_id)
+    }
+
+    /// Deploy a module, returning its [`ModuleId`]. The ID is computed using a
+    /// `blake3` hash of the `bytecode`.
+    /// If contract exports an initialization method, it will be called with
+    /// the arguments provided.
+    ///
+    /// If one needs to specify the ID, [`deploy_with_id`] is available.
+    /// If one would like to *not* call the initialization method, [`deploy`] is
+    /// available.
+    ///
+    /// [`ModuleId`]: ModuleId
+    /// [`deploy_with_id`]: `Session::deploy_with_id`
+    /// [`deploy`]: `Session::deploy`
+    pub fn deploy_and_init<Arg>(
```

Comment:
It's probably better to modify the existing function `deploy` and `deploy_with_id` to take an `Option<Arg>`. Doing it this way implies the caller should somehow "know" that the given bytecode has a constructor,, which they are clearly not in a position to do.